x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -1798,15 +1798,20 @@ static BytesReference createTokenDocument(
                     .startObject("client")
                     .field("type", "unassociated_client")
                     .field("user", originatingClientAuth.getUser().principal())
-                    .field("realm", originatingClientAuth.getAuthenticatedBy().getName())
-                    .endObject()
-                    .endObject();
+                    .field("realm", originatingClientAuth.getAuthenticatedBy().getName());
+                if (originatingClientAuth.getSourceRealm().getDomain() != null) {
+                    builder.field("realm_domain", originatingClientAuth.getSourceRealm().getDomain());
+                }
+                builder.endObject().endObject();
```

Comment:
Existing code capture the `authenticatedBy` realm and I think it's wrong. I have separate PR (#84263) for fix it.